hugr-cli/src/validate.rs

```diff
@@ -33,32 +40,91 @@ pub enum CliError {
     /// Error reading input.
     #[error("Error reading input: {0}")]
     Input(#[from] clap_stdin::StdinError),
+    /// Error reading input.
+    #[error("Error reading from path: {0}")]
+    InputFile(#[from] std::io::Error),
     /// Error parsing input.
     #[error("Error parsing input: {0}")]
     Parse(#[from] serde_json::Error),
     /// Error validating HUGR.
     #[error("Error validating HUGR: {0}")]
     Validate(#[from] hugr_core::hugr::ValidationError),
+    /// Error registering extension.
+    #[error("Error registering extension: {0}")]
+    ExtReg(#[from] hugr_core::extension::ExtensionRegistryError),
+}
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+/// Package of module HUGRs and extensions.
+/// The HUGRs are validated against the extensions.
+pub struct Package {
+    /// Module HUGRs included in the package.
+    pub modules: Vec<Hugr>,
+    /// Extensions to validate against.
+    pub extensions: Vec<Extension>,
+}
+
+impl Package {
+    /// Create a new package.
+    pub fn new(modules: Vec<Hugr>, extensions: Vec<Extension>) -> Self {
+        Self {
+            modules,
+            extensions,
+        }
+    }
 }
 
 /// String to print when validation is successful.
 pub const VALID_PRINT: &str = "HUGR valid!";
 
 impl CliArgs {
     /// Run the HUGR cli and validate against an extension registry.
-    pub fn run(&self, registry: &ExtensionRegistry) -> Result<Hugr, CliError> {
-        let mut hugr: Hugr = serde_json::from_reader(self.input.clone().into_reader()?)?;
-        if self.mermaid {
-            println!("{}", hugr.mermaid_string());
+    pub fn run(&self) -> Result<Vec<Hugr>, CliError> {
+        let rdr = self.input.clone().into_reader()?;
+        let val: serde_json::Value = serde_json::from_reader(rdr)?;
+        // read either a package or a single hugr
+        let (mut modules, packed_exts) = if let Ok(Package {
+            modules,
+            extensions,
+        }) = serde_json::from_value::<Package>(val.clone())
+        {
+            (modules, extensions)
+        } else {
+            let hugr: Hugr = serde_json::from_value(val)?;
+            (vec![hugr], vec![])
+        };
+
+        let mut reg: ExtensionRegistry = if self.no_std {
+            hugr_core::extension::PRELUDE_REGISTRY.to_owned()
+        } else {
+            hugr_core::std_extensions::std_reg()
+        };
```

Comment:
It's a bit counter-intuitive that `no_std` still includes a prelude registry. It should at least be mentioned in the arg docs.